Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -61,29 +67,43 @@ void DataReaderTask::run(ProcessingContext& pc)
   auto device = pc.services().get<o2::framework::RawDeviceService>().device();
   auto outputRoutes = pc.services().get<o2::framework::RawDeviceService>().spec().outputs;
   auto fairMQChannel = outputRoutes.at(0).channel;
-  int inputcount = 0;
+  mDataSpec = o2::header::gDataDescriptionRawData;
+
+  std::vector<InputSpec> dummy{InputSpec{"dummy", ConcreteDataMatcher{"TRD", mDataSpec, 0xDEADBEEF}}};
```

Comment:
not sure what is the point of using hardcoded `gDataDescriptionRawData` here and having the same flexible via `auto inputs = o2::framework::select(inputSpec.c_str())` in the creation of the device. If this "flexible" way is different from `gDataDescriptionRawData` (i.e. RAWDATA), the protection will not work.

Reply:
What about this? If the input can be anything different from the `RAWDATA`, this protection will not work. Do you have such inputs?

Reply:
Its now passed from the DataRead to the the DataReadTask.